src/autoscaler/pruner/metricsdb.go

```diff
@@ -0,0 +1,69 @@
+package pruner
+
+import (
+	"time"
+
+	"autoscaler/db"
+
+	"code.cloudfoundry.org/clock"
+	"code.cloudfoundry.org/lager"
+)
+
+type MetricsDBPruner struct {
+	logger     lager.Logger
+	metricsDB  db.MetricsDB
+	interval   time.Duration
+	cutoffDays int
+	clock      clock.Clock
+	ticker     clock.Ticker
+	doneChan   chan bool
+}
+
+func NewMetricsDBPruner(logger lager.Logger, metricsDB db.MetricsDB, interval int, cutoffDays int, clock clock.Clock) *MetricsDBPruner {
+	return &MetricsDBPruner{
+		logger:     logger,
+		metricsDB:  metricsDB,
+		interval:   time.Duration(interval) * time.Hour,
+		cutoffDays: cutoffDays,
+		clock:      clock,
+		doneChan:   make(chan bool),
+	}
+}
+
+func (mdp *MetricsDBPruner) Start() {
+	mdp.ticker = mdp.clock.NewTicker(mdp.interval)
```

Comment:
why does the ticker live here? It should be scoped to the startMetricPrune which is really pruneMetrics().